policies/sbom/helpers.py

```
@@ -1,5 +1,24 @@
+
+    Accepts either ``"GPL.*,AGPL.*"`` or ``'["GPL.*", "AGPL.*"]'``.
+    Returns a list of stripped, non-empty strings.
+    """
+    raw = raw.strip()
+    if not raw:
+        return []
+    if raw.startswith("["):
+        try:
+            return [p.strip() for p in json.loads(raw) if isinstance(p, str) and p.strip()]
+        except json.JSONDecodeError as e:
+            raise ValueError(f"Invalid JSON array in pattern input: {e}")
```

Ruff (0.15.6)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)